vm-setup/roles/v1aX_integration_test/tasks/node_reuse.yml

```diff
@@ -224,7 +224,7 @@
 
   - name: Update maxSurge and maxUnavailable fields in MachineDeployment.
     shell: |
-        kubectl get machinedeployment "{{ CLUSTER_NAME }}" -n "{{ NAMESPACE }}" -o json | jq '.spec.strategy.rollingUpdate.maxSurge=0|.spec.strategy.rollingUpdate.maxUnavailable=1' | kubectl apply -f-
+        kubectl patch -n "{{ NAMESPACE }}" machinedeployment "{{ CLUSTER_NAME }}" --type=merge -p '{"spec": {"strategy": {"rollingUpdate": [{"maxSurge": 0,"maxUnavailable": 1}]}}}'
```

Comment:
```suggestion
        kubectl patch -n "{{ NAMESPACE }}" machinedeployment "{{ CLUSTER_NAME }}" --type=merge -p '{"spec": {"strategy": {"rollingUpdate": {"maxSurge": 0,"maxUnavailable": 1}}}}'
```

Reply:
It was suggested to use `[]` if patch is for more than a single value, but seems that is not the [case](https://github.com/kubernetes/kubernetes/issues/72663#issuecomment-452325790), we could simply pass them without `[]`? Thanks for the suggestion @lentzi90

Reply:
Hmm I see. I think the suggestion is only for `--type=json` that has a different format. It is very confusing with these `--type` things because all of them takes a JSON-formatted string as input but they work very differently.

For the `merge` type that we are using here it will just use the string to override the resource in the cluster, so the format must be the same. This means that the `rollingUpdate` field should have the same type in the patch as it has in the cluster, and this is a map, not a list. I.e. `kubectl explain machinedeployment.spec.strategy.rollingUpdate` says `RESOURCE: rollingUpdate <Object>` not `<[]Object>`.

I'm not sure if I'm explaining it in a good way, but anyway the point is that it should be without `[]`.

P.S. For `--type=json` it would look something like the below (but I don't think we should use this).
```
kubectl patch ... --type=json '[{ "op": "replace", "path": "/spec/strategy/rollingUpdate/maxSurge", "value": 0 },
    { "op": "replace", "path": "/spec/strategy/rollingUpdate/maxUnavailable", "value": 1 }]'
```